projects/ktor/koin-ktor/src/main/kotlin/org/koin/ktor/plugin/KoinPlugin.kt

```diff
@@ -34,7 +35,8 @@ import org.koin.dsl.KoinAppDeclaration
  */
 val Koin = createApplicationPlugin(name = "Koin", createConfiguration = { KoinApplication.init() }) {
     val koinApplication = setupKoinApplication()
-    startKoin(koinApplication)
+    runningKoinApplication?.let { stopKoin() } // for ktor auto-reload
```

Comment:
You are keeping a ref on the Koin context instance here. Either I would suggest using `KoinPlatformTools.defaultContext().getOrNull()` to request current Koin context and see if it's null